showkase/src/main/java/com/airbnb/android/showkase/ui/CommonComponents.kt

```diff
@@ -63,30 +67,35 @@ internal fun ComponentCard(
     onClick: (() -> Unit)? = null,
     darkMode: Boolean = false,
 ) {
-    val composableModifier = Modifier.generateComposableModifier(metadata)
-    val composableContainerModifier = Modifier.generateContainerModifier(onClick)
-    MaterialTheme(
-        colors = if (darkMode) darkColors() else lightColors()
-    ) {
-        Card(
-            shape = MaterialTheme.shapes.large
+    // This is added to make sure that the navigation of the ShowkaseBrowser does not break
+    // when one of the previews has a back press handler in the implementation of the component.
+    val backPressedDispatcherOwner = rememberOnBackPressedDispatcherOwner()
```

Comment:
Doing this at the top level (which was previously the case) was causing the navigation to not work at all. Instead, I moved this down to specifically happen for the Composables that are responsible for rendering the component previews as that's where we want to override/intercept any back handling that the previews might be doing themselves.